File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    
    stack = []
    for ss in s:
        if len(stack) == 0:
            stack.append(ss)
        
        elif ss == ")" and stack[-1] == "(":
            stack.pop()
        
        else:
            stack.append(ss)
        
    print(stack)
    return len(stack) == 0

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    static boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        
        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (ch == '(') {
                stack.push(ch);
            } else if (ch == ')') {
                if (stack.empty() || stack.peek() != '(') {
                    return false;
                }
                stack.pop();
            }
        }
        
        return stack.empty();
    }
}
